### PR TITLE
Change jdl to don't set nativeLanguage and languages if not present in jdl.

### PR DESCRIPTION
**Pull request by mshima**
Closes https://github.com/jhipster/generator-jhipster/issues/12809.

- try to set nativeLanguage to current locale
- defaults to en when environment variable CI is true

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: test/jdl/jhipster/default-application-options.spec.js [hunk 1]
```diff
@@ -479,17 +479,12 @@ describe('DefaultApplicationOptions', () => {
             it('should set the jhipster prefix option to jhi', () => {
                 expect(options.jhiPrefix).to.equal('jhi');
             });
-            it('should set the languages option to en and fr', () => {
-                expect(options.languages).to.have.lengthOf(2);
-                expect(options.languages).to.include('en');
-                expect(options.languages).to.include('fr');
+            it('should set the languages option to empty', () => {
```
Comment (MathieuAA) [inline test/jdl/jhipster/default-application-options.spec.js:482]: should not set

**Review by MathieuAA — CHANGES_REQUESTED** [review]
So the user has to specify a language if the locale one isn't the right one? all the time?
I don't think it's a good idea...

File: test/jdl/jhipster/default-application-options.spec.js (hunk 1 above)
Comment (mshima) [reply test/jdl/jhipster/default-application-options.spec.js:482]: Empty array then.

**Comment by mshima** [comment]
> So the user has to specify a language if the locale one isn't the right one? all the time?
> I don't think it's a good idea...

Why is better to set 'en' then guessing from user locale?

File: test/jdl/jhipster/default-application-options.spec.js (hunk 1 above)
Comment (mshima) [reply test/jdl/jhipster/default-application-options.spec.js:482]:
```suggestion
            it('should set the languages option to empty array', () => {
```

File: test/jdl/jhipster/default-application-options.spec.js [hunk 2]
```diff
@@ -479,17 +479,12 @@ describe('DefaultApplicationOptions', () => {
...
+                expect(options.languages).to.have.lengthOf(0);
```
Comment (mshima) [inline test/jdl/jhipster/default-application-options.spec.js:483]:
```suggestion
                expect(options.languages).to.be.eql([]);
```

**Comment by MathieuAA** [comment]
Unless the user explicitly asks for a language, getting it fro the OS's locale may not work: my locale's set to French, but my apps are always in English.

**Comment by mshima** [comment]
@MathieuAA dropped user language.

**Comment by mshima** [comment]
> Unless the user explicitly asks for a language, getting it fro the OS's locale may not work: my locale's set to French, but my apps are always in English.

You just need to set your apps native language and languages at jdl.

Having default to user locale should mainly influence at examples like `jhipster jdl bug-tracker`.

**Review by MathieuAA — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
I didn't follow closely what is done here, but in my opinion, the command `jhipster jdl bug-tracker.jh` should generate the same project, no matter the machine

If it's different because of local, we'll have a lot of difficulty to reproduce, to guess the locale of the user etc...

Just my opinion

**Comment by mshima** [comment]
@pascalgrimaud I planning to merge this PR so we can change the generator to make `jhipster jdl bug-tracker.jh --native-language xx` generate only the native language.

> I didn't follow closely what is done here, but in my opinion, the command `jhipster jdl bug-tracker.jh` should generate the same project, no matter the machine
> 
> If it's different because of local, we'll have a lot of difficulty to reproduce, to guess the locale of the user etc...
> 
> Just my opinion

Your concern is valid I will think about this.
Can you repost your comment at https://github.com/jhipster/generator-jhipster/issues/12900?

**Comment by pascalgrimaud** [comment]
fine @mshima : I'm commenting the other ticket

**Pull request merged**